### PR TITLE
Github Actions. Add rubocop and running condition for the ragel pipeline.

### PR DESCRIPTION
**Pull request by alexeevit**
…ndle install in ruby tests

### Description
Please describe your pull request. Thank you for contributing! You're the best.

### Your checklist for this pull request

- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [x] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [x] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [x] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

**Comment by MSP-Greg** [comment]
@alexeevit

Along with the above, MSP-Greg/setup-ruby-pkgs is now ruby/setup-ruby-pkgs

**Comment by alexeevit** [comment]
@MSP-Greg yep, fixed it here and in the ruby tests as well.

I've checked the "paths" filter, works well: https://github.com/alexeevit/puma/pull/2

**Comment by alexeevit** [comment]
I also found that we have a few of flaky specs. Should I create issues about them?

**Comment by MSP-Greg** [comment]
> have a few flaky specs

I assume you mean intermittent test failures/errors?  If you find clear fixes for them, feel free to open a PR, but often they are timing related, and timing can be an issue with shared CI VM's.  Or, most are really intermittent, so it's hard to determine if one has really fixed them.

Conversely, if they really represent a failure in that Puma moves to a state that isn't accounted for, it's a problem.

Sorry, not the best answer.

Have you had a chance to look my comments above?

**Comment by alexeevit** [comment]
> Have you had a chance to look my comments above?

I see only one comment about `ruby/ruby-setup`

> I assume you mean intermittent test failures/errors? If you find clear fixes for them, feel free to open a PR, but often they are timing related, and timing can be an issue with shared CI VM's. Or, most are really intermittent, so it's hard to determine if one has really fixed them.

> Conversely, if they really represent a failure in that Puma moves to a state that isn't accounted for, it's a problem.

> Sorry, not the best answer.

It's a good answer. Yes, I meant intermttent test failures. I'm not sure, but it seems that we don't really care about them. I'd prefer to keep all cases under control. At least, I think it'd be a good idea to have an up-to-date list of all of them. Thus the community could help to investigate and fix them.

**Review by MSP-Greg — CHANGES_REQUESTED** [review]

File: .github/workflows/ragel.yml [hunk 1]
```diff
@@ -1,11 +1,32 @@
...
 
-on: [push, pull_request, workflow_dispatch]
+on:
+  push:
+    paths:
+      - 'ext/**'
+      - '.github/workflows/**'
+  pull_request:
+    paths:
+      - 'ext/**'
```
[the remark this answers is not in the document]
Comment (alexeevit) [reply .github/workflows/ragel.yml:10]: done

File: .github/workflows/ragel.yml [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
+      - uses: MSP-Greg/setup-ruby-pkgs@v1
+        with:
+          ruby-version: 2.6
+          bundler-cache: true # `bundle install` and cache
+      - name: rubocop
+        run: bundle exec rake rubocop ext
   ragel:
     name: >-
       ragel ${{ matrix.os }} ${{ matrix.ruby }}
+    needs: rubocop
```
[the remark this answers is not in the document]
Comment (alexeevit) [reply .github/workflows/ragel.yml:29]: done

**Comment by MSP-Greg** [comment]
> Have you had a chance to look my comments above?

Sorry, I forgot to click one button, I see you've seen them...

> it seems that we don't really care about them

Actually, there's been quite a few small changes trying to fix them.

> At least, I think it'd be a good idea to have an up-to-date list of all of them. Thus the community could help to investigate and fix them.

A list, or maybe add 'intermittent failure/error' to the assert/refute message.  Or both.  I've looked at a few of them, and they're difficult.  Often, one can never get them to fail locally...

**Comment by MSP-Greg** [comment]
@alexeevit

Sorry, I should have clearer.  There were two instances of `- '.github/workflows/**'` in the ragel.yml file.  You changed one to `- '.github/workflows/ragel.yml'.  Can you change the other?  Otherwise, LGTM

**Comment by alexeevit** [comment]
> @alexeevit
> 
> Sorry, I should have clearer. There were two instances of `- '.github/workflows/**'` in the ragel.yml file. You changed one to `- '.github/workflows/ragel.yml'. Can you change the other? Otherwise, LGTM

Oh sorry, I rushed. Fixed.

**Comment by alexeevit** [comment]
> A list, or maybe add 'intermittent failure/error' to the assert/refute message. Or both.

Yep, this could be cool. I'd also add retries for such tests.

> I've looked at a few of them, and they're difficult. Often, one can never get them to fail locally...

Yeah, I see. But this list or tag could increase our awareness of them and we probably should check if they're just bad tests or there're floating bugs.

**Review by MSP-Greg — APPROVED** [review]

**Comment by MSP-Greg** [comment]
Thanks for fixing that last item.

> add retries for such tests

We're already using minitest-retry, not sure if another retry mechanism is needed.

> this list or tag could increase our awareness of them

One problem with the above is that some of the failures seem to be dependent on the Ruby being used (version, 'MRI', platform, etc).  For instance, a failure tolerated on a Ruby head version maybe shouldn't be tolerated on a release version...

> check if they're just bad tests or there're floating bugs.

Hard to say.  Given that CI passes more than not, and also that failures are often just one job, it's messy.  One way to check things is to use the test seed value locally and see if the failure can be repo'd.  I haven't done it for a while, but when I did, the tests often passed.

**Comment by MSP-Greg** [comment]
@alexeevit Thanks!

**Pull request merged**